client/app/views/popover_screens/main.coffee

```diff
@@ -304,26 +315,40 @@ module.exports = class MainPopoverScreen extends PopoverScreenView
         calendarEvent = new Event attrs
         @duplicateButton.hide()
         @spinner.show()
-        calendarEvent.save null,
-            wait: true
-            success: =>
-                @duplicateButton.show()
-                @spinner.hide()
-            error: =>
-                @duplicateButton.show()
-                @spinner.hide()
-
 
-    # Revert formModel to model state, so the close method will not
-    # detect any change.
-    cancelChanges: ->
-        @context.formModel = @model.clone()
-
-
-    # Hides popover.
-    onCancelClicked: ->
-        @cancelChanges()
-        @popover.close()
+        formModelDiffers = not _.isEqual @context.formModel.attributes,
+                                    @model.attributes
+
+        @duplicateButton.show()
```

Comment:
You're right, there is something wrong here. I have to delete lines 322 and 323 here instead because they are already triggered at 344/345. And when we use _updatePopover, there are no duplicate buttons anymore (so doesn't need them). Thanks.